Libraries/LibC/string.cpp

```diff
@@ -457,9 +457,50 @@ char* strpbrk(const char* s, const char* accept)
 
 char* strtok(char* str, const char* delim)
 {
-    (void)str;
-    (void)delim;
-    ASSERT_NOT_REACHED();
+    static char* saved_str;
+    
+    if (!str) {
+        if (!saved_str)
+            return nullptr;
+        str = saved_str;
+    }
+    
+    size_t token_start = 0;
+    size_t token_end = 0;
+    
+    for (size_t i = 0; i < strlen(str); ++i) {
+        bool is_delim = false;
+        for (size_t j = 0; j < strlen(delim); ++j) {
+            if (str[i] == delim[j]) {
+                is_delim = true;
+                ++token_start;
+                break;
+            }
+        }
+        if (!is_delim) break;
+    }
+    
+    if (strlen(&str[token_start]) == 0)
```

Comment:
Instead of wasting time computing the length of the entire string here, we only need to find out if the string is empty.
We can do that by comparing the first character with '\0' instead :)